### PR TITLE
[Master] Updated recovery API docs

### PR DESCRIPTION
**Pull request by divyaamunugama**
Related issue: https://github.com/wso2/product-is/issues/16584

### CODE REVIEW

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 1]
```diff
@@ -468,27 +737,42 @@ sms notification templates.
```
Comment (Rashmini) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:737]:
```suggestion
2. Use the `recoveryCode` and a channel `id` to get notifications via that channel.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 2]
```diff
@@ -468,27 +737,42 @@ sms notification templates.
...
+                }
+            ]
+        }"
         ```
         ---
         **Sample Request**
         ```curl
         {
-        "code": "PWR-02001",
+        "code": "PWR-02002",
```
Comment (Rashmini) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:761]:
```suggestion
        "code": "PWR-02001",
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 3]
```diff
@@ -514,24 +811,42 @@ sms notification templates.
...
                 "type": "POST"
             }
         ]
         }
         ```
 
-4. Use the `confirmationCode` to verify the password reset.
+4. Use the `flowConfirmationCode` recieved in the previous response as the `confirmationCode` in this request to verify the password reset.
+
+    !!! note "OTP parameter"
```
Comment (Rashmini) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:828]: This note is not required for EXTERNAL.

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 4]
```diff
@@ -514,24 +811,42 @@ sms notification templates.
...
-                "href": "/t/carbon.super/api/users/v1/recovery/password/resend",
+                "href": "/t/carbon.super/api/users/v2/recovery/password/resend",
                 "type": "POST"
             }
         ]
         }
         ```
 
-4. Use the `confirmationCode` to verify the password reset.
+4. Use the `flowConfirmationCode` recieved in the previous response as the `confirmationCode` in this request to verify the password reset.
```
Comment (Rashmini) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:826]:
```suggestion
4. Use the `confirmationCode` to verify the password reset.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 5]
```diff
@@ -514,24 +811,42 @@ sms notification templates.
...
-        curl -X POST "https://localhost:9443/api/users/v1/recovery/password/confirm" -H "Authorization: Basic YWRtaW46YWRtaW4=" -H "accept: application/json" -H "Content-Type: application/json" -d "{\"confirmationCode\":\"1234-2ws34-12345\",\"properties\":[{\"key\":\"key\",\"value\":\"value\"}]}"
+        ```curl
+        curl -X POST "https://localhost:9443/api/users/v2/recovery/password/confirm" 
+        -H "Authorization: Basic YWRtaW46YWRtaW4=" 
+        -H "accept: application/json" 
+        -H "Content-Type: application/json" 
+        -d "
+        {
+            "confirmationCode":"1234-2ws34-12345",
+            "otp":"gt4k1d" # This parameter is required only if you are using OTP based recovery. Remove this if you are using Email based recovery.
```
Comment (Rashmini) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:842]: Not required.

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 6]
```diff
@@ -562,4 +892,140 @@ sms notification templates.
...
+                    "type": "POST"
+                }
+            ]
+            }]
+            ```
+
+        !!! note
+                The validity period of the recovery code is determined by the [2nd step of configuring password recovery](#password-recovery-flow)
+
+    2. Use the `recoveryCode` and a channel `id` to get the recovered username.
```
Comment (Rashmini) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:933]:
```suggestion
    2. Use the `recoveryCode` and a channel `id` to get notifications via that channel.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 7]
```diff
@@ -377,24 +438,42 @@ sms notification templates.
...
             }
         ]
         }
         ```
 
-4. Use the confirmation code received by the user to verify the confirmation code.
+4. Verify the  confirmation code/OTP received.
+
+    !!! note "OTP parameter"
+        - It is required to add the OTP parameter to the request if you are using OTP based recovery. Use the `flowConfirmationCode` received in the previous response as the `confirmationCode` in this request
```
Comment (Rashmini) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:456]:
```suggestion
        - It is required to add the OTP parameter to the request if you are using OTP based recovery. Use the `flowConfirmationCode` received in the previous response as the `confirmationCode` in this request.
```

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 8]
```diff
@@ -404,19 +483,33 @@ sms notification templates.
...
+                    "href": "/t/carbon.super/api/users/v2/recovery/password/reset",
                     "type": "POST"
                 }
             ]
         }
         ```
 
 5. Use the `resetCode` and the new password to update the existing password and recover the account.
-
+5. Use the `resetCode`, `flowConfirmationCode` and the new password to update the existing password and recover the account.
```
Comment (Rashmini) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:494]: This should be the line replacing line 493.

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 9]
```diff
@@ -465,30 +734,45 @@ sms notification templates.
             The validity period of the recovery code is determined by the [2nd step of configuring password recovery](#password-recovery-flow)
 
 2. Use the `recoveryCode` and a channel `id` to get the recovered username.
-
+2. Use the `recoveryCode` and a channel `id` to get notifications via that channel.
```
Comment (Rashmini) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:737]: This should be the line replacing line 736

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 10]
```diff
@@ -404,19 +483,33 @@ sms notification templates.
...
                     "type": "POST"
                 }
             ]
         }
         ```
 
-5. Use the `resetCode` and the new password to update the existing password and recover the account.
-
+5. Use the `resetCode`, `flowConfirmationCode` and the new password to update the existing password and recover the account.
     !!! abstract ""
```
Comment (Rashmini) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:494]: Add new line

File: en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md [hunk 11]
```diff
@@ -464,31 +733,45 @@ sms notification templates.
     !!! note
             The validity period of the recovery code is determined by the [2nd step of configuring password recovery](#password-recovery-flow)
 
-2. Use the `recoveryCode` and a channel `id` to get the recovered username.
-
+2. Use the `recoveryCode` and a channel `id` to get notifications via that channel.
     !!! abstract ""
```
Comment (Rashmini) [inline en/docs/guides/identity-lifecycles/extend-username-and-password-recovery.md:737]: Add new line

**Review by Rashmini — APPROVED** [review]

**Review by himeshsiriwardana — APPROVED** [review]

**Pull request merged**